enhancements/network/egress-ip-per-destinations.md

```diff
@@ -0,0 +1,327 @@
+
+As a cluster administrator, I want to easily configure which destination networks should have an `EgressIP` because the destination
+networks may change, and I do not wish to have to update individual `EgressIP` custom resources for every update.
+
+### Goals
+- Allow a set of pods to have a consistent source IP depending on the destination
+- Support shared and local gateway (aka `routingViaHost=true`) modes in-order to reduce asymmetry between the modes
+- Backward compatibility with existing `EgressIP` API
+- If the new API isn't defined, preserve existing behaviour
+
+### Non-Goals
+- Change existing `EgressIP` behavior when the new field is not defined
+
+## Proposal
+
+A new `TrafficSelector` field is added to `EgressIP` spec which uses Kubernetes label selector semantics
+to select a new CRD called `EgressIPTraffic` which contains a list of destination networks. This list of networks defines when an `EgressIP` should
+be used as source IP when an application attempts to talk to a destination address, and it is within one of the predefined networks
+defined within a `EgressIPTraffic` CR.
+
+The reason for not embedding the destination networks within each `EgressIP` CR is that it places a burden
+on the cluster administrator to update all `EgressIP` CRs if the destination networks change.
+
+With the Kubernetes label selector semantics, cluster administrators can define a set of destination networks that multiple `EgressIP`
+CRs can consume.
+
+If greater than one `EgressIPTraffic` CRs are selected, the networks are added together, and if a destination IP is within one
+of the networks, then the pods source IP will be the `EgressIP`.
+If `TrafficSelector` is not specific, existing `EgressIP` behaviour is conserved.
+
+### Workflow Description
+
+**cluster admin** creates a set of namespaces for tenants to use.
+Tenant has requested that its applications need a consistent source IP
+for 2 different traffic paths - one health signal traffic and one for work load traffic.
+The cluster admin updates a dedicated egress node by attaching two new networks on separate interfaces.
+The cluster admin labels this node egress-able.
+The cluster admin creates two `EgressIPTraffic` CRs, specifies a destination network in each and labels each CR differently.
+The cluster admin creates two `EgressIP` CRs each containing an IP that falls within the previously added networks range.
+The two `EgressIP`s selects the same set of pods but different `TrafficSelector`s to select the previously created `EgressIPTraffic` CRs.
+The `EgressIP`s are automatically assigned to the secondary host interfaces on the egress node.
+
+The cluster admin gives permissions to the **application admin** to the namespaces.
+
+The **application admin** deploys its application and the application has a consistent but different source IP for the
+health signal traffic and the work load traffic.
+
+### API Extensions
+
+Add `TrafficSelector` field to `EgressIPSpec`:
```

Comment:
Where is this exist (?) type defined?

Reply:
`go` type definition:
https://github.com/openshift/ovn-kubernetes/tree/master/go-controller/pkg/crd/egressip/v1